File: ios/Sources/VaultSync/Domain/Repositories/DocumentRepository.swift
import Foundation

public protocol DocumentRepository: Sendable {
    func observeDocuments() -> AsyncStream<[Document]>
    func getDocuments() async throws -> [Document]
    func getDocument(id: String) async throws -> Document?
    func saveDocument(_ document: Document) async throws
    func deleteDocument(id: String) async throws
    func getDocumentsByStatus(_ status: SyncStatus) async throws -> [Document]
    func updateSyncStatus(id: String, status: SyncStatus) async throws
    func findBySha256(_ sha256: String) async throws -> Document?
}

File: android/app/src/main/java/com/vaultsync/domain/repository/DocumentRepository.kt
package com.vaultsync.domain.repository

import com.vaultsync.domain.model.Document
import com.vaultsync.domain.model.SyncStatus
import kotlinx.coroutines.flow.Flow

interface DocumentRepository {
    fun observeDocuments(): Flow<List<Document>>
    suspend fun getDocuments(): List<Document>
    suspend fun getDocument(id: String): Document?
    suspend fun saveDocument(document: Document)
    suspend fun deleteDocument(id: String)
    suspend fun getDocumentsByStatus(status: SyncStatus): List<Document>
    suspend fun updateSyncStatus(id: String, status: SyncStatus)
    suspend fun findBySha256(sha256: String): Document?
}
